x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/custom/CustomServiceSettings.java

```diff
@@ -162,24 +147,44 @@ public static TextEmbeddingSettings fromMap(Map<String, Object> map, TaskType ta
             SimilarityMeasure similarity = extractSimilarity(map, ModelConfigurations.SERVICE_SETTINGS, validationException);
             Integer dims = removeAsType(map, DIMENSIONS, Integer.class);
             Integer maxInputTokens = removeAsType(map, MAX_INPUT_TOKENS, Integer.class);
-            return new TextEmbeddingSettings(similarity, dims, maxInputTokens, DenseVectorFieldMapper.ElementType.FLOAT);
```

Comment:
The element type logic has been delegated to the `TextEmbeddingResponseParser` so removing it from being hard coded to float here.